admin/class-social-facebook.php

```diff
@@ -29,8 +36,47 @@ public function __construct() {
 	 *
 	 */
 	public function show_form() {
-		$form = new Yoast_Social_Facebook_Form();
-		$form->show_form();
+		$this->form->show_form();
+	}
+
+	/**
+	 * Adding a new admin
+	 * @param string $admin_name
+	 * @param string $admin_id
+	 */
+	public function add_admin( $admin_name, $admin_id ) {
+		// If one of the fields is empty
+		if ( empty( $admin_name ) || empty( $admin_id ) ) {
+			return json_encode(
+				array(
+					'success' => 0,
+					'html'    => "<p class='notice-error notice'><span style='margin-left: 5px'>" . __( 'Be sure both requested fields are filled properly.', 'wordpress-seo' ) . '</span></p>',
+				)
+			);
+		}
+
+		// Clean up the url part
+		$admin_id = trim( parse_url( $admin_id, PHP_URL_PATH ), '/' );
+
+		if ( ! isset( $this->options['fb_admins'][ $admin_id ] ) ) {
+			$this->options['fb_admins'][ $admin_id ]['name'] = sanitize_text_field( urldecode( $admin_name ) );
+			$this->options['fb_admins'][ $admin_id ]['link'] = sanitize_text_field( urldecode( 'http://www.facebook.com/' . $admin_id ) );
+
+			$this->save_options();
+
+			$return = array(
+				'success' => 1,
+				'html'    => $this->form->get_admin_link( $admin_id, $this->options['fb_admins'][ $admin_id ] ),
+			);
+		}
+		else {
+			$return = array(
+				'success' => 0,
+				'html'    => "<p class='notice-error notice'><span style='margin-left: 5px'>" . __( 'This Facebook user is already added as an admin.', 'wordpress-seo' ) . '</span></p>',
```

Comment:
I'd change this sentence to `This Facebook user has already been added as an admin.` (present perfect) because it happened in the past (not time specific), and the effect still lasts.